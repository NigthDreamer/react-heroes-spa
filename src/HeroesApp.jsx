import { AppRouter } from './router/AppRouter';

export const HeroesApp = () => {
  //No se puede meter un provider por encima del Approuter
  return <AppRouter />;
};
