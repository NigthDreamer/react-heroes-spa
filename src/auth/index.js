export * from "./pages";
export * from "./context";
export * from "./types";
