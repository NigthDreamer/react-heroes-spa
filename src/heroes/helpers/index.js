export * from './getHeroById';
export * from './getHeroesByName';
export * from './getHeroesByPublisher';
export * from './toPathRelative';
