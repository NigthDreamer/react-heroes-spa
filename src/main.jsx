import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import { HeroesApp } from './HeroesApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <HeroesApp />
  </React.StrictMode>
);
