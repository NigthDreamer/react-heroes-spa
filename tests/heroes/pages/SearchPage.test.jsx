/* eslint-disable no-undef */
import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes';

//Creo el mock del useNavigate
const mockedUseNavigate = jest.fn();

/**
 * Mockeo la librería donde se encuentra el useNavigate. OJO!!! La funcion anonima
 * devuelve un jsx con un objeto dentro, por eso los parentesis y despues las llaves
 */
jest.mock('react-router-dom', () => ({
  //Devuelvo todas las funciones de la librería
  ...jest.requireActual('react-router-dom'),
  /**
   * Expecto la funcion del hook que vamos a usar, la cual vamos a sobreescribir con
   * Nuestro mock
   */
  useNavigate: () => mockedUseNavigate,
}));

describe('Pruebas en <SearchPage/>', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Debe de mostrarse correctamente con valores por defecto', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrarse a Batman y el input con el valor de queryString', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox');
    expect(input.value).toBe('batman');

    const img = screen.getByRole('img');
    expect(img.src).toContain('dc-batman.jpg');

    const div = screen.getByTestId('search-div');
    expect(div.style.display).toBe('none');
  });

  test('Debe de mostrar un error si no se encuentra el hero (batman123)', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman123']}>
        <SearchPage />
      </MemoryRouter>
    );

    const error = screen.getByTestId('search-error-div');
    expect(error.style.display).not.toBe('none');
  });

  test('Debe de llamar al navigate al hacer el submit del formulario', () => {
    const inputValue = 'superman';

    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    );

    //Seteamos el valor del input
    const input = screen.getByRole('textbox');
    fireEvent.change(input, {
      target: { name: 'searchText', value: inputValue },
    });

    //Hacemos el submit del formulario
    const form = screen.getByRole('form');
    fireEvent.submit(form);
    screen.debug();

    expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);
  });
});
