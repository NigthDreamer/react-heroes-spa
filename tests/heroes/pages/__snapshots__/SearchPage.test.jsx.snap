// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage/> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-3"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <div
        class="alert alert-primary mt-3 animate__animated animate__fadeIn"
        data-testid="search-div"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger mt-3 animate__animated animate__fadeIn"
        data-testid="search-error-div"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
